docs/docs/02-guide/06-tokenfactory/01-tokenfactory.md

```diff
@@ -0,0 +1,622 @@
+
+```proto
+message MsgUpdateDenom {
+  string owner = 1;
+  string denom = 2;
+  string description = 3;
+  string url = 4;
+  int32 maxSupply = 5;
+  bool canChangeMaxSupply = 6;
+}
+```
+
+### Client Logic
+
+In the `x/tokenfactory/client/cli/tx_denom.go` file, update the client application logic.
+
+**For `CmdCreateDenom`:**
+
+- Adjust the number of arguments from 8 to 7, removing references to the supply argument, and update the usage descriptions.
+
+**For `CmdUpdateDenom()`:**
+
+- Reduce the number of arguments to 5, excluding `supply`, `precision`, and `ticker`, and modify the usage descriptions accordingly.
+
+Also, update the tests in `x/tokenfactory/client/cli/tx_denom_test.go` to reflect these changes.
+
+### Types Updates
+
+When creating new denoms, they initially have no supply. The supply is determined only when tokens are minted.
+
+In `x/tokenfactory/types/messages_denom.go`:
+
+- Remove the `supply` parameter from `NewMsgCreateDenom`.
+- Update `NewMsgUpdateDenom` to exclude unchangeable parameters like `ticker`, `precision`, and `supply`.
+
+Implement basic input validation in `x/tokenfactory/types/messages_denom.go`:
+
+- Ensure the ticker length is between 3 and 10 characters.
+```go
+func (msg *MsgCreateDenom) ValidateBasic() error {
+    _, err := sdk.AccAddressFromBech32(msg.Owner)
+    if err != nil {
+        return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid owner address (%s)", err)
+    }
+
+    tickerLength := len(msg.Ticker)
+    if tickerLength < 3 {
+        return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "Ticker length must be at least 3 chars long")
+    }
+    if tickerLength > 10 {
+        return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "Ticker length must be 10 chars long maximum")
+    }
+    if msg.MaxSupply == 0 {
+        return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "Max Supply must be greater than 0")
+    }
+    
+    return nil
+}
+```
+
+- Set `maxSupply` to be greater than 0.
+
+```go
+func (msg *MsgUpdateDenom) ValidateBasic() error {
+    _, err := sdk.AccAddressFromBech32(msg.Owner)
+    if err != nil {
+        return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid owner address (%s)", err)
+    }
+    if msg.MaxSupply == 0 {
+        return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "Max Supply must be greater than 0")
+    }
+    return nil
+}
+```
+
+### Keeper Logic
+
+The keeper is where you define the business logic for manipulating the database and writing to the key-value store.
+
+**In `x/tokenfactory/keeper/msg_server_denom.go`:**
+
+- Update `CreateDenom()` to include logic for creating unique denoms. Modify the error message to point to existing denoms. Set `Supply` to `0`.
+- Modify `UpdateDenom()` to verify ownership and manage max supply changes.
+
+```go
+func (k msgServer) UpdateDenom(goCtx context.Context, msg *types.MsgUpdateDenom) (*types.MsgUpdateDenomResponse, error) {
+    ctx := sdk.UnwrapSDKContext(goCtx)
```

Comment:
I think we don't need to wrap and unwrap context anymore in the new version of the sdk